geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlHandler.java

```diff
@@ -122,32 +115,28 @@ private String getKeyColumnName(Connection connection, String tableName) {
   }
 
   PdxInstance executeReadStatement(PreparedStatement statement, List<ColumnValue> columnList,
-      PdxInstanceFactory factory, RegionMapping regionMapping, String keyColumnName) {
+      PdxInstanceFactory factory, RegionMapping regionMapping, String keyColumnName)
+      throws SQLException {
     PdxInstance pdxInstance = null;
-    try {
-      setValuesInStatement(statement, columnList);
-      try (ResultSet resultSet = statement.executeQuery()) {
-        if (resultSet.next()) {
-          ResultSetMetaData metaData = resultSet.getMetaData();
-          int ColumnsNumber = metaData.getColumnCount();
-          for (int i = 1; i <= ColumnsNumber; i++) {
-            Object columnValue = resultSet.getObject(i);
-            String columnName = metaData.getColumnName(i);
-            String fieldName = mapColumnNameToFieldName(columnName, regionMapping);
-            if (regionMapping.isPrimaryKeyInValue()
-                || !keyColumnName.equalsIgnoreCase(columnName)) {
-              factory.writeField(fieldName, columnValue, Object.class);
-            }
-          }
-          if (resultSet.next()) {
-            throw new IllegalStateException(
-                "Multiple rows returned for query: " + resultSet.getStatement().toString());
+    setValuesInStatement(statement, columnList);
+    try (ResultSet resultSet = statement.executeQuery()) {
+      if (resultSet.next()) {
+        ResultSetMetaData metaData = resultSet.getMetaData();
+        int ColumnsNumber = metaData.getColumnCount();
+        for (int i = 1; i <= ColumnsNumber; i++) {
+          Object columnValue = resultSet.getObject(i);
+          String columnName = metaData.getColumnName(i);
+          String fieldName = mapColumnNameToFieldName(columnName, regionMapping);
+          if (regionMapping.isPrimaryKeyInValue() || !keyColumnName.equalsIgnoreCase(columnName)) {
+            factory.writeField(fieldName, columnValue, Object.class);
           }
-          pdxInstance = factory.create();
         }
+        if (resultSet.next()) {
+          throw new IllegalStateException(
```

Comment:
Seems like this should be JdbcConnectorException

Reply:
Since we are using the JdbcConnectorException to help users with catching/handling causes of issues in the connector, I think you are right.